src/sinks/elasticsearch.rs

```diff
@@ -88,37 +89,48 @@ struct ElasticSearchCommon {
     credentials: Option<AwsCredentials>,
 }
 
+#[derive(Debug, Snafu)]
+enum ParseError {
+    #[snafu(display("Invalid host {:?}: {:?}", host, source))]
+    InvalidHost { host: String, source: InvalidUri },
+    #[snafu(display("AWS provider requires a configured region"))]
+    AWSRequiresRegion,
+    #[snafu(display("Could not create AWS credentials provider: {:?}", source))]
+    AWSCredentialsProviderFailed { source: CredentialsError },
+    #[snafu(display("Could not generate AWS credentials: {:?}", source))]
+    AWSCredentialsGenerateFailed { source: CredentialsError },
+}
+
 impl ElasticSearchCommon {
-    fn parse_config(config: &ElasticSearchConfig) -> Result<Self, String> {
+    fn parse_config(config: &ElasticSearchConfig) -> Result<Self, crate::Error> {
```

Comment:
probably makes sense to expose a `crate::Result` type that is `type Result<T, E = crate::Error> = std::result::Result<T, E>;`

Reply:
Probably